internal/controller/registry.go

```diff
@@ -57,19 +57,21 @@ type Registry struct {
 	independentSyncPeriod time.Duration
 	featureFlags          *featureflags.FeatureFlags
 
-	logger          *zap.Logger
-	reconcilers     []Reconciler
-	globalSecretRef client.ObjectKey
+	logger                  *zap.Logger
+	reconcilers             []Reconciler
+	globalSecretRef         client.ObjectKey
+	experimentalReconcilers bool
 }
 
-func NewRegistry(predicates []predicate.Predicate, deletionProtection bool, logger *zap.Logger, independentSyncPeriod time.Duration, featureFlags *featureflags.FeatureFlags, globalSecretRef client.ObjectKey) *Registry {
+func NewRegistry(predicates []predicate.Predicate, deletionProtection bool, logger *zap.Logger, independentSyncPeriod time.Duration, featureFlags *featureflags.FeatureFlags, globalSecretRef client.ObjectKey, experimentalReconcilers bool) *Registry {
 	return &Registry{
-		sharedPredicates:      predicates,
-		deletionProtection:    deletionProtection,
-		logger:                logger,
-		independentSyncPeriod: independentSyncPeriod,
-		featureFlags:          featureFlags,
-		globalSecretRef:       globalSecretRef,
+		sharedPredicates:        predicates,
+		deletionProtection:      deletionProtection,
+		logger:                  logger,
+		independentSyncPeriod:   independentSyncPeriod,
+		featureFlags:            featureFlags,
+		globalSecretRef:         globalSecretRef,
+		experimentalReconcilers: experimentalReconcilers,
```

Comment:
```suggestion
		isExperimentalEnabled: isExperimentalEnabled,
```

Reply:
I think I will even remove this config bit.